packages/block-editor/src/components/off-canvas-editor/block.js

```diff
@@ -389,16 +393,18 @@ function ListViewBlock( {
 													'core/navigation-submenu',
 													block.attributes,
 													block.innerBlocks
-														? [
-																...block.innerBlocks,
-																newLink,
-														  ]
-														: [ newLink ]
 												);
+
 												replaceBlock(
 													clientId,
 													newSubmenu
 												);
+
+												replaceInnerBlocks(
+													newSubmenu.clientId,
+													[ newLink ],
+													updateSelectionOnInsert
+												);
```

Comment:
We'll need a comment here but essentially by making a seperately call to update the inner blocks with the newly created custom link, it triggers the `getLastInsertedBlockClientId` selector which then ensures the Link UI is shown.

Reply:
@getdave  What makes the Link UI show by calling `getLastInsertedBlockClientId`?

Reply:
OK, I was not updated and couldn't find the call :) I found them in the effect checking `BLOCKS_WITH_LINK_UI_SUPPORT` 🆗

Reply:
There's an effect which runs when the `getLastInsertedBlockClientId` selector changes. This sets the Link UI to show.  See https://github.com/WordPress/gutenberg/pull/46744